pandas/core/indexes/numeric.py

```diff
@@ -328,7 +328,9 @@ def astype(self, dtype, copy=True):
             msg = ('Cannot convert Float64Index to dtype {dtype}; integer '
                    'values are required for conversion').format(dtype=dtype)
             raise TypeError(msg)
-        elif is_integer_dtype(dtype) and self.hasnans:
+        elif (is_integer_dtype(dtype) and
```

Comment:
Did you add a test for this? 
(it is currently failing (meaning: raising an error) on master)

Reply:
this is well tested in many locations

Reply:
Ah, yes, forgot about the object dtype of the result